### PR TITLE
Release checklist

### PR DESCRIPTION
**Pull request by gunamata**
Checklist to follow during the release process.

### CODE REVIEW

**Review by mook-as — COMMENTED** [review]
Looks fine to me, but I'd much rather defer to @mattfarina — he is actually involved in the release process.

File: docs/development/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## Release Checklist
+
+- [ ] Update version number in package.json if not done after last release.
+- [ ] Tag release branch. Wait for the CI to build artifacts.
+- [ ] Sign windows installer.
+- [ ] Sign Linux installer.
```
Comment (mattfarina) [inline docs/development/release-checklist.md:6]: We don't need to sign Linux anymore. This is automated by OBS.

File: docs/development/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- [ ] git clean, reset to make sure a clean (CI equivalent) build.
+- [ ] Manually zip the installer.
+- [ ] Rename installer filename to replace space with dot.
+
+### Release Documentation
+- [ ] Release notes. Update on the Github draft Release page.
+- [ ] docs update (Help, Readme..)
+- [ ] Slack Announcements
+- [ ] Newsletter summary
+- [ ] Update metrics, roadmap on Confluence page
```
Comment (mattfarina) [inline docs/development/release-checklist.md:18]: These are internal things. Should that be in the public release material? I wouldn't put internal things in here.

File: docs/development/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+### Release  
+- [ ] Perform smoke test on release artifacts.
+- [ ] Upload Linux, mac, win release artifacts on the Github draft Release page.
```
Comment (mattfarina) [inline docs/development/release-checklist.md:22]: The Linux artifacts are now automated. When a release it cut OBS will pick it up and create the release artifacts. The Linux element can be removed from here. A new bullet should be added to make sure the linux repos are in the release notes.

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: docs/development/release-checklist.md (hunk 1 above)
Comment (gunamata) [reply docs/development/release-checklist.md]: Removed `Linux` statement.

File: docs/development/release-checklist.md (hunk 2 above)
Comment (gunamata) [reply docs/development/release-checklist.md:18]: Removed marketing section.

File: docs/development/release-checklist.md (hunk 3 above)
Comment (gunamata) [reply docs/development/release-checklist.md]: Updated the statement to remove Linux element.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**